model/test_results.go

```diff
@@ -186,40 +199,38 @@ func (t *TestResults) Append(ctx context.Context, results []TestResult) error {
 		return nil
 	}
 
-	bucket, err := t.GetBucket(ctx)
-	if err != nil {
-		return err
-	}
-
-	var allResults testResultsDoc
-	r, err := bucket.Get(ctx, testResultsCollection)
+	allResults, err := t.downloadBSON(ctx)
 	if err != nil && !pail.IsKeyNotFoundError(err) {
 		return errors.Wrap(err, "getting uploaded test results")
 	}
-	if err == nil {
-		catcher := grip.NewBasicCatcher()
+	allResults = append(allResults, results...)
 
-		data, err := ioutil.ReadAll(r)
-		catcher.Add(r.Close())
-		if err != nil {
-			catcher.Wrap(err, "reading uploaded test results")
-			return catcher.Resolve()
-		}
-
-		if err = bson.Unmarshal(data, &allResults); err != nil {
-			catcher.Wrap(err, "unmarshalling uploaded test results")
-			return catcher.Resolve()
-		}
+	// TODO (EVG-16140): Stop calling this once we do the BSON to Parquet
+	// cutover.
+	if err := t.uploadBSON(ctx, allResults); err != nil {
+		return errors.Wrap(err, "appending BSON test results")
 	}
-	allResults.Results = append(allResults.Results, results...)
 
-	data, err := bson.Marshal(&allResults)
+	bucket, err := t.GetPrestoBucket(ctx)
 	if err != nil {
-		return errors.Wrap(err, "marshalling test results")
+		return err
 	}
-
-	if err := bucket.Put(ctx, testResultsCollection, bytes.NewReader(data)); err != nil {
-		return errors.Wrap(err, "uploading test results")
+	w, err := bucket.Writer(ctx, t.PrestoPartitionKey())
+	if err != nil {
+		return errors.Wrap(err, "creating Presto bucket writer")
+	}
+	pw, err := writer.NewParquetWriterFromWriter(w, new(ParquetTestResults), 1)
+	if err != nil {
+		return errors.Wrap(err, "creating new Parquet writer")
+	}
+	if err = pw.Write(t.convertToParquet(allResults)); err != nil {
+		return errors.Wrap(err, "writing Parquet test results")
+	}
+	if err = pw.WriteStop(); err != nil {
+		return errors.Wrap(err, "stopping Parquet writer")
+	}
+	if err := w.Close(); err != nil {
```

Comment:
Should this be closed in a defer on L222?

Reply:
I am going to add a defer func with a conditional because I think it is worth keeping this here since `w.Close()` will flush the data to S3 and if that errors we should definitely know about it.